### PR TITLE
doc: Add workflow documentation

### PR DESCRIPTION
**Pull request by marikaner**
## What Has Changed?

I added a simple use case for the workflow usage.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [ ] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [ ] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  - OpenAPI
+  - REST
+  - JavaScript
+  - TypeScript
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+The [SAP Workflow service](https://help.sap.com/viewer/product/WORKFLOW_SERVICE/Cloud/en-US) allows you to automate business processes across organizations and applications on SAP Business Technology Platform, both Cloud Foundry and Neo environment.
+It is technically separated into two services per environment: the Inbox API (OData) and the Workflow API (OpenAPI).
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:18]: I think a bulleted list looks nicer after a colon.

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+The Inbox API is mainly meant to build a personal inbox, while the Workflow API helps to manage workflows and tasks across recipients.
+
+The first SAP Cloud SDK pregenerated OpenAPI client is for the OpenAPI Workflow API.
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:21]: Open**API** client is for the Open**API** Workflow **API**

At least the second `OpenAPI` could be removed as it's implied...

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+The client for the Workflow API is available as an npm package.
+To install it, run:
+
+```bash
+$ npm install @sap/cloud-sdk-workflow-service-cf
+```
+
+## Use the Client
+
+For this exemplary use case, assume a simple workflow definition:
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:46]:
```suggestion
For this example, assume following workflow definition:
```

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+<img src={useBaseUrl('img/workflow-definition.png')} style={{ width: '50%' }} />
+
+It contains one task definition to approve a request through a form with one boolean field, `approve`.
+The following use case shows how to initiate a workflow, retrieve its tasks, complete a task and retrieve the execution logs for this workflow.
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:51]:
```suggestion
The use case shows how to initiate a workflow, retrieve its tasks, complete a task and retrieve the execution logs.
```

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+): Promise<TaskInstance> {
+  const [taskInstance] = await UserTaskInstancesApi.queryInstances({
+    workflowInstanceId: workflowInstance.id
+  }).execute(destination);
+
+  return taskInstance;
+}
+```
+
+The function above returns an instance of a task, that looks as follows:
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:112]:
```suggestion
The function above returns an instance of a task.
```

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+It has properties like the id of the instance and definition (`id` and `definitionId`), the `status`, `priority`, etc.
+
+### Update the Status and Context of a Task
+
+To update a task, use the `updateInstance` on the `UserTaskInstancesApi`.
+Pass the mandatory instance id as first parameter.
+The second parameter is the body to be sent with the update.
+Is has the type `UpdateTaskInstancePayload` and allows you to pass the parameters `context` and `status`, among others.
+In this example the task requires to give (or not to give) an approval.
+The form to fulfill this task has one boolean field, `approve`:
```
Comment (florian-richter) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx:148]:
```suggestion
The form to fulfill this task has one boolean field, `approve`.
```

File: sidebars.js [hunk 7]
```diff
@@ -156,7 +156,12 @@ module.exports = {
             items: [
               'js/features/openapi/overview',
               'js/features/openapi/generate-openapi-client',
-              'js/features/openapi/execute-openapi-request'
+              'js/features/openapi/execute-openapi-request',
+              {
+                type: 'category',
+                label: 'Pre-generated Clients',
```
Comment (florian-richter) [inline sidebars.js:162]:
```suggestion
                label: 'Pregenerated Clients',
```

**Review by florian-richter — CHANGES_REQUESTED** [review]
Please consider my points as suggestions.

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx (hunk 6 above)
Comment (marikaner) [reply docs/js/features/openapi/pre-generated-clients/workflow.mdx:148]: I like it if there is a connection to images or code. Without the colon I would have to reference it in the text somehow. Why do you think a fullstop is better here?

File: docs/js/features/openapi/execute-openapi-request.mdx [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
 sidebar_label: Execute an OpenAPI Request
-description: This article describes how to generate a client from the a service definition and how to use the client.
+description: How to use a client (pre-)generated by the SAP Cloud SDK OpenAPI generator.
 keywords:
   - sap
   - cloud
   - sdk
   - OpenAPI
-  - rest
+  - REST
```
[the remark this answers is not in the document]
Comment (marikaner) [reply docs/js/features/openapi/execute-openapi-request.mdx:10]: I can also change it back. Somehow "rest" just didn't make sense to me. I don't know too exactly what this is relevant for anyways.

File: sidebars.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (marikaner) [reply sidebars.js]: Fixed it, I will also add a rule in the next PR

File: docs/js/features/openapi/generate-openapi-client.mdx [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Use the OpenAPI Generator to Generate Typed Clients
 sidebar_label: Generate an OpenAPI Client
-description: This article describes how generate a client from the a service definition and how to use the client.
+description: How to generate a client from a service definition.
```
[the remark this answers is not in the document]
Comment (marikaner) [reply docs/js/features/openapi/generate-openapi-client.mdx:4]: I reverted my change, I don't think it is important.

File: docs/js/features/openapi/pre-generated-clients/workflow.mdx [hunk 10]
```diff
@@ -0,0 +1,225 @@
+---
+title: Use the Pregenerated Workflow API Client for Cloud Foundry
+sidebar_label: SAP Workflow API for Cloud Foundry
+description: This article describes what the Workflow API for Cloud Foundry is and how
```
Comment (marikaner) [inline docs/js/features/openapi/pre-generated-clients/workflow.mdx]:
```suggestion
description: This article describes what the Workflow API for Cloud Foundry is and how to consume it with the SAP Cloud SDK.
```

File: docs/js/features/openapi/pregenerated-clients/workflow.mdx [hunk 11]
```diff
@@ -0,0 +1,229 @@
+---
+title: Use the Pregenerated Workflow API Client for Cloud Foundry
+sidebar_label: SAP Workflow API for Cloud Foundry
+description: This article describes what the Workflow API for Cloud Foundry is and how
```
Comment (marikaner) [inline docs/js/features/openapi/pregenerated-clients/workflow.mdx:4]:
```suggestion
description: This article describes what the Workflow API for Cloud Foundry is and how to consume it with the SAP Cloud SDK.
```

File: docs/js/features/openapi/overview.mdx [hunk 12]
```diff
@@ -42,7 +34,7 @@ Another way to get a client library is to [generate it yourself](generate-openap
 
 ## Why Using SAP Cloud SDK for JavaScript With OpenAPI Services?
 
-- You'll benefit from type-safety, less code boilerplate, and better developer experience.
+- You'll benefit from less code boilerplate, and better developer experience.
```
Comment (florian-richter) [inline docs/js/features/openapi/overview.mdx:37]:
```suggestion
- You'll benefit from less code boilerplate and better developer experience.
```
I think the Oxford comma is not needed if there are only 2 options, right?

File: docs/js/features/openapi/pregenerated-clients/workflow.mdx [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+
+:::note Prerequisites
+The following documentation explains how to use the pregenerated client for:
+
+- An existing instance of the SAP Workflow service
+- A user, who has the necessary roles to make the requests
+- An existing workflow definition that contains one simple user task
+
+To learn how to set up your environment for the SAP Workflow service, see the [Workflow service documentation](https://help.sap.com/viewer/e157c391253b4ecd93647bf232d18a83/Cloud/en-US/fc3d44872c354742afd672aa8d9c16b4.html).
+:::
```
Comment (florian-richter) [inline docs/js/features/openapi/pregenerated-clients/workflow.mdx:37]:
```suggestion
:::note Prerequisites

The following documentation explains how to use the pregenerated client for:

- An existing instance of the SAP Workflow service
- A user, who has the necessary roles to make the requests
- An existing workflow definition that contains one simple user task

To learn how to set up your environment for the SAP Workflow service, see the [Workflow service documentation](https://help.sap.com/viewer/e157c391253b4ecd93647bf232d18a83/Cloud/en-US/fc3d44872c354742afd672aa8d9c16b4.html).

:::
```

**Review by florian-richter — APPROVED** [review]
LGTM

**Review by jjtang1985 — APPROVED** [review]
LGTM

**Pull request merged**